query/decode.ts

```diff
@@ -62,10 +62,15 @@ function decodeBinary() {
   throw new Error("Decoding binary data is not implemented!");
 }
 
-function decodeText(value: Uint8Array, typeOid: number) {
+function decodeText(value: Uint8Array, typeOid: number, decoders?: Decoders) {
```

Comment:
Have you considered using `type DecoderFunction = (value: string, oid: number) => unknown;` instead of `Decoders`? I wonder if that might be better to implement, maintain and use.

Reply:
I think it'll complicate handling the Oids that users don't override in the function. with `Decoders` the lib will know exactly which types to override and which to use the default behavior.